From 5fe26f3ba7acd5d84d5010b746ca6a4618fccc6a Mon Sep 17 00:00:00 2001
From: Adrian Perez de Castro <aperez@igalia.com>
Date: Thu, 15 Aug 2024 13:41:45 +0300
Subject: [PATCH] [WPE][GTK] Crash in
 WebCore::TextDecorationPainter::paintBackgroundDecorations when compiled with
 Clang with LTO enabled https://bugs.webkit.org/show_bug.cgi?id=274780

Reviewed by NOBODY (OOPS!).

Clang seem to have some issue when inlining assignment and move
operators in LTO builds, generating code that tries to perform OOB
access to Vector data. Replacing an assignmenmt with a Vector::swap(),
which is semantically equivalent in this case (the moved-from object
is not used again in the function) workarounds the compiler issue.

* Source/WebCore/rendering/TextDecorationPainter.cpp:
(WebCore::translateIntersectionPointsToSkipInkBoundaries):
---
 Source/WebCore/rendering/TextDecorationPainter.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Source/WebCore/rendering/TextDecorationPainter.cpp b/Source/WebCore/rendering/TextDecorationPainter.cpp
index 5c4e798d7aef..6c3951145f30 100644
--- a/Source/WebCore/rendering/TextDecorationPainter.cpp
+++ b/Source/WebCore/rendering/TextDecorationPainter.cpp
@@ -141,7 +141,7 @@ static DashArray translateIntersectionPointsToSkipInkBoundaries(const DashArray&
                 intermediateTuples.append(*i);
         }
     } else
-        intermediateTuples = tuples;
+        intermediateTuples.swap(tuples);
 
     // Step 3: Output the space between the ranges, but only if the space warrants an underline.
     float previous = 0;
